### PR TITLE
fix(http): resolve `withRequestsMadeViaParent` behavior with `withFetch`

### PR DESCRIPTION
**Pull request by alan-agius4**
This commit addresses dependency injection defects when using the `withFetch` API. Formerly, utilizing `withFetch` led to the automatic setting of `HttpBackend` to `FetchBackend`, which proved problematic in certain scenarios. Notably, conflicts arose when integrating `withRequestsMadeViaParent` and manually overriding tokens, as observed in instances like `InMemoryWebApiModule`.

**Reviewer note:** I suggest that you enable `hide whitespace`.

### CODE REVIEW

File: packages/common/http/src/provider.ts [hunk 1]
```diff
@@ -294,26 +298,13 @@ export function withRequestsMadeViaParent(): HttpFeature<HttpFeatureKind.Request
 /**
  * Configures the current `HttpClient` instance to make requests using the fetch API.
  *
- * This `FetchBackend` requires the support of the Fetch API which is available on all evergreen
```
Comment (alan-agius4) [inline packages/common/http/src/provider.ts:297]: These are no longer needed as with v18 all versions of Node.js support fetch

File: packages/common/http/test/provider_spec.ts [hunk 2]
```diff
@@ -472,20 +487,33 @@ describe('provideHttpClient', () => {
       expect(consoleWarnSpy.calls.count()).toBe(0);
     });
 
-    it('withFetch should always override the backend', () => {
+    it(`'withFetch' should not override provided backend`, () => {
```
Comment (JeanMeche) [inline packages/common/http/test/provider_spec.ts:490]: If I remember well this was one of the feature we wanted for the FetchBackend. Did we change our mind on this ?

Comment (alan-agius4) [reply packages/common/http/test/provider_spec.ts:490]: No, it is just that this was a bit flawed. The original feature was that if withFetch is used the Fetch HttpBackend will be used for the current and child injectors in the case of having `provideHttpClient` in a sub module. I have added a test for this.

However, overridden the backend should still be possible as in some cases it is needed.

**Review by AndrewKushnir — APPROVED** [review]

**Review by JeanMeche — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 9ddb003b6cab080662bacaf93417bd599357700b.

**Pull request closed without merging**